js/web/components/EmailCapture.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useCallback, useEffect, useContext } from 'react'
+import Button from '@mui/material/Button'
+import Typography from '@mui/material/Typography'
+import Modal from '@mui/material/Modal'
+import { styled } from '@mui/material/styles'
+import * as Yup from 'yup'
+import EmailCaptureForm from './EmailCaptureForm'
+import { Box } from '@mui/material'
+import { useWallet } from '@solana/wallet-adapter-react'
+import Nina from '@nina-protocol/nina-internal-sdk/esm/Nina'
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  p: 4,
+}
+const ITEM_HEIGHT = 48
+const ITEM_PADDING_TOP = 8
+
+const EmailCaptureSchema = Yup.object().shape({
+  email: Yup.string().email().required('Email is Required'),
+  soundcloud: Yup.string(),
+  twitter: Yup.string(),
+  instagram: Yup.string(),
+  wallet: Yup.string(),
+  type: Yup.string(),
+})
+
+const EmailCapture = ({ size }) => {
+  const { publicKey, connected } = useWallet()
+  const { submitEmailRequest } = useContext(Nina.Context)
+  const [open, setOpen] = useState(false)
+  const handleOpen = () => setOpen(true)
+  const handleClose = () => setOpen(false)
+  const [formValues, setFormValues] = useState({})
+  const [formIsValid, setFormIsValid] = useState(false)
+
+  useEffect(() => {
+    if (connected) {
+      setFormValues({ ...formValues, wallet: publicKey.toString() })
+    }
+  }, [connected, publicKey])
+
+  const submitAndCloseModal = () => {
+    handleSubmit()
+    handleClose()
+  }
+
+  const handleSubmit = async () => {
+    console.log('submit')
+    console.log('formValues', formValues)
+    console.log('EmailCaptureSchema', EmailCaptureSchema)
+    const isValid = await EmailCaptureSchema.isValid(formValues)
+
+    if (isValid) {
+      submitEmailRequest(formValues)
+    } else {
+      console.log('invalid')
+    }
+  }
+
+  const handleFormChange = useCallback(
+    async (values) => {
+      const newValues = { ...formValues, ...values }
+      const isValid = await EmailCaptureSchema.isValid(newValues)
+      setFormIsValid(isValid)
+      setFormValues(newValues)
+    },
+    [formValues]
+  )
+
+  return (
+    <div>
+      {size === 'large' && (
```

Comment:
different layouts across devices should be handled by media queries. Unless there is a reason for using a prop to drive this I dont think we want to introduce this pattern